Oracle-Validation.md

```diff
@@ -0,0 +1,73 @@
+# Client-Side Oracle Validation
+
+# Table of Contents
+
+* [Abstract](#abstract)
+* [General Requirements](#general-requirements)
+* [Oracle-Discovery](#oracle-discovery)
+* [Oracle-Validation](#oracle-validation)
+
+
+## Abstract
+
+The DLC protocol relies on trust-minized oracles mapping real-world events to signatures, those ones
+then confidentialy re-mapped by participants to a range of contract outcomes. A participant expected
+execution of the DLC is thus function of the oracle well-behaving as announced before event
+realization. In this current protocol version, oracle behavior is assumed to be trusted. This level
+of trust is expressed by the client oracle selection policy, and should faithfully reflect user
+oracle preferences.
+
+Oracle signature usage requires confidence that the event associated public key is owned by the
+correct remote subject (person, system or organization). The ownership is asserted by having the
+client verifying a signature which is bound to an identity anchor (domain name, LN node pubkey, ...).The evaluation of identity anchor correctness is beyond the scope of this protocol. A DLC client may
+use this identity anchor as a descriptor for a reputation system.
+
+# Oracle Discovery
+
+Actually, no oracle discovery protocol support is mandated. The following requirements are generic
+requirements, which should be swayed from only if protocol has higher built-in privacy mechanisms.
+
+## Requirements
+
+A node:
+* MUST fetch oracle pubkey.
+* MUST fetch any oracle pubkey available.
+* MUST persist its keyring across restart/shutdown.
+
```

Comment:
Should we also recommend a encrypted & authenticated channel so they have less of chance of being man-in-the-middle'd

Reply:
Also should we add recommendations of where to obtain the oracle's keys? ie directly from oracle vs oracle explorer

Reply:
Yes I think rn we don't have specific communication protocols for oracle discovery, you might have
* a DNS seed (e.g BOLT 10) relying on trusting DNS servers
* a manual copy-paste from few oracle repositories (e.g 1ml.com)
* a oracle existence gossips network (e.g BOLT 7) with some attributes filtered by user preferences
* a hardcoded initial list provided by DLC client vendors
* ....

Even if you answer "What's a good oracle to me ?", you still have to find the list of such "good" oracles without them being hidden or unknown from you, or if you have some central repo avoiding this one becoming a bait ?

Reply:
> Should we also recommend a encrypted & authenticated channel so they have less of chance of being man-in-the-middle'd

I don't see the point of an authenticated channel to receive oracle information if the oracle announcements (and attestations) are themselves authenticated.

 > Also should we add recommendations of where to obtain the oracle's keys? ie directly from oracle vs oracle explorer

I think it's up to the users who they trust more, and I don't really see a technical rational for choosing one over the other? (and thus I don't think we should recommend anything, at least at this stage).

Reply:
> I don't see the point of an authenticated channel to receive oracle information if the oracle > announcements (and attestations) are themselves authenticated.

It _might_ be good to heartbeat your connections with a oracle to detect any kind of oracle censorship attempt. Wouldn't achieve its effect with unauthenticated channel. Though such heartbeat might not be great to preserve your privacy toward the oracle, trade-off...

> I think it's up to the users who they trust more, and I don't really see a technical rational for choosing one over the other? (and thus I don't think we should recommend anything, at least at this stage).

Once we have lazyness/fraud proofs we might have some kind of scoring and selection engine? But overall agree too early for now..

Reply:
> It might be good to heartbeat your connections with a oracle to detect any kind of oracle censorship attempt. Wouldn't achieve its effect with unauthenticated channel. Though such heartbeat might not be great to preserve your privacy toward the oracle, trade-off...

Yeah not great for privacy, and I don't really see a big upside. If you get censored there is not much you can do I think? And if there is something you can do, you should be able to do it even if you detect it only at maturation time if you setup a long enough refund time lock.

> Once we have lazyness/fraud proofs we might have some kind of scoring and selection engine? But overall agree too early for now..

Yes we might. But still I personally feel it will be up to the users which scoring or selection engine they'd want to chose, or if they just trust an oracle more than those. I think what I'm saying is that such considerations might not have their place in the specifications.

Reply:
>  If you get censored there is not much you can do I think? And if there is something you can do, you should be able to do it even if you detect it only at maturation time if you setup a long enough refund time lock.

First we should dissociate oracle censorship from bitcoin p2p censorship. They may overlap or not. Secondly, you might have redundant communication channels, only enabled in case of emergency (including worst-case scenario sending a mail to a human operator to get involved). And if it's a tx-relay censorship, it might be hard to dissociate from a congestioned mempool...

That said, no a worry to address those issues now.